### ISSUE TITLE
`ext:save-lisp-and-die` breaks since 2.2.0

### ISSUE THREAD

**Issue by bohonghuang** [clasp-developers/clasp#1432]
**Describe the bug**
Saving the image appears to be successful, but it crashes when booting Clasp with the saved image:

```
COMMON-LISP-USER> (ext:save-lisp-and-die "/tmp/clasp" :executable t)
Total object memory usage (bytes):    166557968
Total GC_get_heap_size():             688168960
Total GC_get_free_bytes():            370421760
Total GC_get_bytes_since_gc():                0
Total GC_get_total_bytes():         21419841891
!
!
!
! Passed the memory test with zero corrupt objects!
! In preparation for snapshot_save.
!
!
!
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2655:snapshot_save Updated with noStomp forwarding for snapshot_save
Running 2 sys:*save-hooks*
Trying to close foreign libraries
Finished closing foreign libraries
makunbound for *jit-pid* and *jit-log-stream*
makunbound for *jit-pid*
makunbound for *jit-log-stream*
Finished makunbound of some symbols
Finished cmp:invoke-save-hooks
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2664:snapshot_save Finished invoking cmp:invoke-save-hooks
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2216:snapshot_save_impl
   size = 206389528
   general_count = 896607
   cons_count = 670215
   weak_count = 2
   ObjectFileCount = 2376
   CodeCount = 2376
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2030:prepareRelocationTableForSave  Dealing with library#0:  /usr/bin/iclasp @0x5626ea5b0500
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2031:prepareRelocationTableForSave  Number of pointers before extracting unique pointers: 1179035
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2045:prepareRelocationTableForSave  Number of unique pointers: 13371
 13000 remaining pointers to dladdr
 12000 remaining pointers to dladdr
 11000 remaining pointers to dladdr
 10000 remaining pointers to dladdr
  9000 remaining pointers to dladdr
  8000 remaining pointers to dladdr
  7000 remaining pointers to dladdr
  6000 remaining pointers to dladdr
  5000 remaining pointers to dladdr
  4000 remaining pointers to dladdr
  3000 remaining pointers to dladdr
  2000 remaining pointers to dladdr
  1000 remaining pointers to dladdr
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2049:prepareRelocationTableForSave  Library #0 contains 13371 grouped pointers
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2055:prepareRelocationTableForSave  Done with library: /usr/bin/iclasp @0x5626ea5b0500
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2030:prepareRelocationTableForSave  Dealing with library#1:  /usr/lib/libclang-cpp.so.15 @0x5626ea5b05b0
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2031:prepareRelocationTableForSave  Number of pointers before extracting unique pointers: 38
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2045:prepareRelocationTableForSave  Number of unique pointers: 38
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2049:prepareRelocationTableForSave  Library #1 contains 38 grouped pointers
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2055:prepareRelocationTableForSave  Done with library: /usr/lib/libclang-cpp.so.15 @0x5626ea5b05b0
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2030:prepareRelocationTableForSave  Dealing with library#2:  /usr/lib/libLLVM-15.so @0x5626ea5b0660
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2031:prepareRelocationTableForSave  Number of pointers before extracting unique pointers: 180
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2045:prepareRelocationTableForSave  Number of unique pointers: 120
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2049:prepareRelocationTableForSave  Library #2 contains 120 grouped pointers
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2055:prepareRelocationTableForSave  Done with library: /usr/lib/libLLVM-15.so @0x5626ea5b0660
Loaded
                  size_t _Magic -> 348235823
     uintptr_t _LibrariesOffset -> 4096(0x1000)
   uintptr_t _NumberOfLibraries -> 3
 uintptr_t _SaveTimeMemoryAddress -> 0x7fe1698ab010
         uintptr_t _MemoryStart -> 2574744(0x274998)
     uintptr_t _NumberOfObjects -> 1566824
          uintptr_t _MemorySize -> 206430968(0xc4de2f8)
         size_t _LispRootOffset -> 205933352
          size_t _LispRootCount -> 1
      size_t _SymbolRootsOffset -> 205933376
       size_t _SymbolRootsCount -> 2586
     uintptr_t _ObjectFileStart -> 209005712(0xc752c90)
      uintptr_t _ObjectFileSize -> 142619264(0x8803280)
       NextUnshiftedClbindStamp -> 67348(0x10714)
             NextUnshiftedStamp -> 134916(0x20f04)
Writing snapshot to /tmp/ss-XXss3TyM filedes = 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 4096 bytes from 0x5626f5b00c00 to filedes 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 2570648 bytes from 0x5626f882cc00 to filedes 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 206430968 bytes from 0x7fe1698ab010 to filedes 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 142619264 bytes from 0x7fe1610a7010 to filedes 5
Creating binary object from snapshot...
Link command:
/usr/bin/clang++ -L/usr/lib -fuse-ld=gold -pthread -fvisibility=default -rdynamic -L/usr/lib/clasp/ -o/tmp/clasp /tmp/ss-XXN3pRdz -Wl,-whole-archive -lclasp -Wl,-no-whole-archive -ldl -lclang-cpp -lLLVM-15 -lfmt -lgmpxx -lgmp -lelf
Linking executable...
```

```
Starting clasp-boehmprecise-2.2.0 from embedded snapshot
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.3
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.4
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.34
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@GLIBC_2.34
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.3
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1977:loadSymbols Could not resolve address with loopup.lookupSymbol or dlsym for symbol _ZZN10asttooling23initialize_clangToolingEvEN3$_08__invokeERN5clang7tooling9ClangToolE
Segmentation fault (core dumped)

```

**Code at issue**

```lisp
(ext:save-lisp-and-die "/tmp/clasp" :executable t)
```

**Context**
* Clasp version: 2.2.0
* LLVM/Clang version: 15.0.7
* Glibc version: 2.37

**Comment by bohonghuang** [comment]
> I am no longer able to reproduce with v2.2.0-64

I still got the same error in the latest build of the Git commit ad35b11a9509a226ea8081343149c505ae820b88:

```
Starting clasp-boehmprecise-2.2.0-68-gad35b11a9 from base image
Loading resource file /home/coco24/.clasprc
Top level in: #<PROCESS TOP-LEVEL @0x7f56b43e1f09 (Running)>.
COMMON-LISP-USER> (ext:save-lisp-and-die "/tmp/clasp" :executable t)
Total object memory usage (bytes):    146767120
Total GC_get_heap_size():             275705856
Total GC_get_free_bytes():             58138624
Total GC_get_bytes_since_gc():                0
Total GC_get_total_bytes():           727963106
!
!
!
! Passed the memory test with zero corrupt objects!
! In preparation for snapshot_save.
!
!
!
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2655:snapshot_save Updated with noStomp forwarding for snapshot_save
Running 2 sys:*save-hooks*
Trying to close foreign libraries
Finished closing foreign libraries
makunbound for *jit-pid* and *jit-log-stream*
makunbound for *jit-pid*
makunbound for *jit-log-stream*
Finished makunbound of some symbols
Finished cmp:invoke-save-hooks
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2664:snapshot_save Finished invoking cmp:invoke-save-hooks
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2216:snapshot_save_impl
   size = 178735784
   general_count = 710415
   cons_count = 547151
   weak_count = 2
   ObjectFileCount = 1327
   CodeCount = 1327
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2030:prepareRelocationTableForSave  Dealing with library#0:  /usr/bin/iclasp @0x556e44206010
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2031:prepareRelocationTableForSave  Number of pointers before extracting unique pointers: 973271
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2045:prepareRelocationTableForSave  Number of unique pointers: 13102
 13000 remaining pointers to dladdr
 12000 remaining pointers to dladdr
 11000 remaining pointers to dladdr
 10000 remaining pointers to dladdr
  9000 remaining pointers to dladdr
  8000 remaining pointers to dladdr
  7000 remaining pointers to dladdr
  6000 remaining pointers to dladdr
  5000 remaining pointers to dladdr
  4000 remaining pointers to dladdr
  3000 remaining pointers to dladdr
  2000 remaining pointers to dladdr
  1000 remaining pointers to dladdr
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2049:prepareRelocationTableForSave  Library #0 contains 13102 grouped pointers
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2055:prepareRelocationTableForSave  Done with library: /usr/bin/iclasp @0x556e44206010
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2030:prepareRelocationTableForSave  Dealing with library#1:  /usr/lib/libclang-cpp.so.15 @0x556e442060c0
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2031:prepareRelocationTableForSave  Number of pointers before extracting unique pointers: 38
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2045:prepareRelocationTableForSave  Number of unique pointers: 38
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2049:prepareRelocationTableForSave  Library #1 contains 38 grouped pointers
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2055:prepareRelocationTableForSave  Done with library: /usr/lib/libclang-cpp.so.15 @0x556e442060c0
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2030:prepareRelocationTableForSave  Dealing with library#2:  /usr/lib/libLLVM-15.so @0x556e44206170
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2031:prepareRelocationTableForSave  Number of pointers before extracting unique pointers: 180
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2045:prepareRelocationTableForSave  Number of unique pointers: 120
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2049:prepareRelocationTableForSave  Library #2 contains 120 grouped pointers
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:2055:prepareRelocationTableForSave  Done with library: /usr/lib/libLLVM-15.so @0x556e44206170
Loaded
                  size_t _Magic -> 348235823
     uintptr_t _LibrariesOffset -> 4096(0x1000)
   uintptr_t _NumberOfLibraries -> 3
 uintptr_t _SaveTimeMemoryAddress -> 0x7f568637d010
         uintptr_t _MemoryStart -> 2527336(0x269068)
     uintptr_t _NumberOfObjects -> 1257568
          uintptr_t _MemorySize -> 178777224(0xaa7ec88)
         size_t _LispRootOffset -> 178481016
          size_t _LispRootCount -> 1
      size_t _SymbolRootsOffset -> 178481040
       size_t _SymbolRootsCount -> 2586
     uintptr_t _ObjectFileStart -> 181304560(0xace7cf0)
      uintptr_t _ObjectFileSize -> 120881320(0x73480a8)
       NextUnshiftedClbindStamp -> 67348(0x10714)
             NextUnshiftedStamp -> 134752(0x20e60)
Writing snapshot to /tmp/ss-XXCEKXCT filedes = 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 4096 bytes from 0x556e4e885920 to filedes 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 2523240 bytes from 0x556e51563010 to filedes 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 178777224 bytes from 0x7f568637d010 to filedes 5
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1195:write_to_filedes Wrote 120881320 bytes from 0x7f567f034010 to filedes 5
Creating binary object from snapshot...
Link command:
/usr/bin/clang++ -L/usr/lib -fuse-ld=gold -pthread -fvisibility=default -rdynamic -L/usr/lib/clasp/ -o/tmp/clasp /tmp/ss-XX57Lhr8 -Wl,-whole-archive -lclasp -Wl,-no-whole-archive -ldl -lclang-cpp -lLLVM-15 -lfmt -lgmpxx -lgmp -lelf
Linking executable...

 
Starting clasp-boehmprecise-2.2.0-68-gad35b11a9 from embedded snapshot
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.3
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.4
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.34
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@GLIBC_2.34
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.3
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.2.5
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_2.35
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:229:loadLibrarySymbolLookup We encountered a symbol that does not have the correct version "@@LLVM_xxx" - it has "@@GLIBC_PRIVATE
" - we need to generalize this version test
/usr/share/clasp/src/gctools/snapshotSaveLoad.cc:1977:loadSymbols Could not resolve address with loopup.lookupSymbol or dlsym for symbol _ZZN10asttooling23initialize_clangToolingEvEN3$_08__invokeERN5clang7tooling9ClangToolE
zsh: segmentation fault (core dumped)  ./clasp
```

**Comment by drmeister** [comment]
Thanks for your patience everyone.   I needed a couple of days focus to fix this one.   Objects that are in hashtables are assigned a 32bit random value called a `badge`. The badge needs to be saved in the snapshot and restored on load.  I recently made the badge atomic because of a race condition.  That broke snapshot save/load of badges.

I fixed it in the fixsnapshot branch and I'm running the tests and then I'll merge it into main.
